File: ProjetoPadraoDotnetCore/Aplication/Models/Request/Template/TemplateRequest.cs
using System.Diagnostics.CodeAnalysis;
using Infraestrutura.Enum;

namespace Aplication.Models.Request.Template;

public class TemplateRequest
{
    public int? IdTemplate { get; set; }
    public string? Titulo { get; set; }
    public string? Descricao { get; set; }
    public string? Foto { get; set; }
    public EEscala? Escala { get; set; }
    public int? QuantidadeTotal { get; set; }
    public int? IdTemplateCategoria { get; set; }
    public string? DescricaoCategoriaNova { get; set; }
    public int? IdUsuarioCadastro { get; set; }
    public List<AtividadeTemplateRequest>? LAtividade { get; set; }
}

public class AtividadeTemplateRequest
{
    public int? TempoPrevisto { get; set; }
    public int? IdTemplate { get; set; }
    public string? Titulo { get; set; }
    public List<TarefaTemplateRequest>? LTarefaTemplate { get; set; }
    public int? Posicao { get; set; }
}

public class TarefaTemplateRequest
{
    public string? Descricao { get; set; }
    public string? DescricaoTarefa { get; set; }
    public EPrioridadeTarefa? Prioridade { get; set; }
    public List<string>? LTagsTarefa { get; set; }
}

File: ProjetoPadraoAngular/src/app/objects/Template/TemplateRequest.ts
export interface TemplateRequest
{
    IdTemplate: number | undefined,
    Titulo: string;
    Descricao: string;
    QuantidadeTotal: number;
    IdTemplateCategoria: number;
    Escala: number,
    DescricaoCategoriaNova: string | undefined,
    IdUsuarioCadastro: number,
    LAtividade: Array<AtividadeTemplateRequest>,
    Foto: string | undefined,
}

export interface AtividadeTemplateRequest
{
    TempoPrevisto: number,
    IdTemplate: number | undefined,
    Titulo: string,
    Posicao: number,
    LTarefaTemplate: Array<TarefaTemplateRequest>
}

export interface  TarefaTemplateRequest
{
    Descricao: string;
    DescricaoTarefa: string;
    Prioridade: number,
    LTagsTarefa: Array<string>
}

